packages/core/botpress/src/util.js

```diff
@@ -123,6 +127,52 @@ const getPackageName = pkg => {
   }
 }
 
+const versionExists = version => {
+  if (version == null) {
+    throw new Error("Version doesn't exist in botfile.js")
+  }
+
+  return version
+}
+const isValidString = version => {
+  if (_.isEmpty(version) || !_.isString(version)) {
+    throw new Error('Version must be non-empty string')
+  }
+
+  return version
+}
+const isValidFormat = version => {
+  try {
+    // TODO: change this method if "semver" module will implement semver.isValid()
+    semver.valid(version)
+  } catch (err) {
+    throw new Error('Version must have semver format (e.g. 10.25.0)')
+  }
+
+  return version
+}
+const isEqualOrGreater = packageVersion => botfileVersion => {
+  const botfileMajorVersion = Number(semver.major(botfileVersion))
+  const packageMajorVersion = Number(semver.major(packageVersion))
+
+  const msg = `Your bot may be incompatible with botpress v${packageVersion}
+  because it looks like your bot was originally created with botpress v${botfileVersion}.
+  To address this:
```

Comment:
Sorry for this taking so much and me probably not being explicit enough. We already know which of these 3 items is the case, we can only display the specific message to the user. Let me update your PR with the change I have in mind